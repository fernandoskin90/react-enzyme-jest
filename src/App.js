import React, { Component } from 'react';
// import logo from './logo.svg';
// import './App.css';

const title = ({ text }) => <div>{ text }</div>

class App extends Component {
  state = {
    on: false,
    input: '',
    mainColor: 'blue',
    lifeCycle: ''
  }
  componentDidMount() {
    this.setState({ lifeCycle: 'componentDidMount' })
  }
  componentWillReceiveProps() {
    this.setState({ lifeCycle: "componentWillReceiveProps" });
  }

  handleStrings(str) {
    return str ? true : false
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to React</h1>
          <h3 className={this.state.mainColor}>Hello!</h3>
        </header>
        <ul>
          <li>h</li>
          <li>h</li>
          <li>h</li>
        </ul>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <p className="button-state">{this.state.on ? 'Yes!' : '!No' }</p>
        <button onClick={() => this.setState({ on: true })}>Click</button>
        <title text='some string' />
        <h2>{this.state.input}</h2>
        <input type="text" onChange={(e) => this.setState({input: e.target.value})}/>
        <p className="lifeCycle">{this.state.lifeCycle}</p>
      </div>
    );
  }
}

export class Link extends Component {
  render(){
    return this.props.hide ? null :  <a href={this.props.address}>Click</a>
  }
}

export default App;
