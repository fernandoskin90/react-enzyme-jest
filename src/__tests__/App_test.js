import React from 'react';
import ReactDOM from 'react-dom';
import App, { Link } from '../App';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json'

describe("<App /> shallow renderig", () => {
  it("should contain 1 p element", () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find("h1").text()).toBe("Welcome to React");
  });
  it('matches the snapshots', () => {
    const tree = shallow(<App />)
    expect(toJson(tree)).toMatchSnapshot()
  });
  it('updates className with new state', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('.blue').length).toBe(1);
    expect(wrapper.find(".red").length).toBe(0);
    // set state
    wrapper.setState({ mainColor: 'red'});
    expect(wrapper.find(".blue").length).toBe(0);
    expect(wrapper.find(".red").length).toBe(1);
  });
  it('on button click changes p text', () => {
    const wrapper = shallow(<App />);
    const button = wrapper.find('button');
    expect(wrapper.find('.button-state').text()).toBe('!No');
    button.simulate('click');
    expect(wrapper.find('.button-state').text()).toBe('Yes!')
  });
  it('On input change, title change text', () => {
    const wrapper = shallow(<App />);
    const input = wrapper.find('input');
    expect(wrapper.find('h2').text()).toBe('');
    input.simulate('change', { target: { value: 'Fernando' }});
    expect(wrapper.find('h2').text()).toBe('Fernando');
  });
});

describe('<Link />', () => {
  it('link component accepts addrss prop', () => {
    const wrapper = shallow(<Link address="www.google.com.co"/>);
    expect(wrapper.instance().props.address).toBe("www.google.com.co")
  });
  it('a tag node renders href correctly', () => {
    const wrapper = shallow(<Link address="www.google.com.co" />)
    expect(wrapper.props().href).toBe("www.google.com.co");
  });
  it('returns null with true hide prop', () => {
    const wrapper = shallow(<Link hide={false} />);
    expect(wrapper.find('a').length).toBe(1);
    wrapper.setProps({ hide: true });
    expect(wrapper.get(0)).toBeNull()
  });
});

// test for lifecycle
describe("<App /> mount renderig", () => {
  it("should contain 1 p element", () => {
    const wrapper = mount(<App />);
    expect(wrapper.find("h1").text()).toBe("Welcome to React");
    wrapper.unmount()
  });
  it("matches the snapshots", () => {
    const tree = mount(<App />);
    expect(toJson(tree)).toMatchSnapshot();
    tree.unmount()
  });
});
