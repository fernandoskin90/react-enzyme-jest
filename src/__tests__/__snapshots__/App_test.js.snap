// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount renderig matches the snapshots 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
      <h3
        className="blue"
      >
        Hello!
      </h3>
    </header>
    <ul>
      <li>
        h
      </li>
      <li>
        h
      </li>
      <li>
        h
      </li>
    </ul>
    <p
      className="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <p
      className="button-state"
    >
      !No
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <title
      text="some string"
    />
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
    <p
      className="lifeCycle"
    >
      componentDidMount
    </p>
  </div>
</App>
`;

exports[`<App /> shallow renderig matches the snapshots 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <h3
      className="blue"
    >
      Hello!
    </h3>
  </header>
  <ul>
    <li>
      h
    </li>
    <li>
      h
    </li>
    <li>
      h
    </li>
  </ul>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <p
    className="button-state"
  >
    !No
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
  <title
    text="some string"
  />
  <h2 />
  <input
    onChange={[Function]}
    type="text"
  />
  <p
    className="lifeCycle"
  >
    componentDidMount
  </p>
</div>
`;
